### PR TITLE
db for ml models (wip)

### PR DESCRIPTION
**Pull request by michaelmattig**
- [ ] I added an entry to [`CHANGELOG.md`](CHANGELOG.md) if knowledge of this change could be valuable to users.

---

Here is a brief summary of what I did:

<TEXT>

### CODE REVIEW

**Comment by michaelmattig** [comment]
[ML-Model-DB](https://trello.com/c/UReOmLzH/1707-ml-model-db)

File: datatypes/src/machine_learning.rs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+use serde::{de::Visitor, Deserialize, Serialize};
+
+use crate::{
+    dataset::{is_invalid_name_char, SYSTEM_NAMESPACE},
+    raster::RasterDataType,
+};
+
+const NAME_DELIMITER: char = ':';
+
+// TODO: custom serialization as ns:name
```
Comment (jdroenner) [inline datatypes/src/machine_learning.rs:12]: you did that, right?

File: operators/src/pro/machine_learning/metadata_from_file.rs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+use geoengine_datatypes::{machine_learning::MlModelMetadata, raster::RasterDataType};
+use snafu::{ensure, ResultExt};
+
+use crate::pro::machine_learning::error::{
+    InvalidInputDimensions, InvalidOutputDimensions, MultipleInputsNotSupported, Ort,
+};
+
+use super::MachineLearningError;
+
+// TODO: extract metadata during model import and load it from database here instead of accessing the model file here
```
Comment (jdroenner) [inline operators/src/pro/machine_learning/metadata_from_file.rs:12]: do you still intend to do this?

File: services/src/machine_learning/name.rs [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+            namespace,
+            name: name.into(),
+        }
+    }
+}
+
+impl From<MlModelName> for geoengine_datatypes::machine_learning::MlModelName {
+    fn from(name: MlModelName) -> Self {
+        Self {
+            namespace: name.namespace.clone(),
```
Comment (jdroenner) [inline services/src/machine_learning/name.rs:37]: why is that clone needed?

**Review by jdroenner — COMMENTED** [review]
how did you solve the type conversion of the model output?

File: datatypes/src/machine_learning.rs (hunk 1 above)
Comment (michaelmattig) [reply datatypes/src/machine_learning.rs]: yes, I removed the comment 07374b850dd9a0d5638233a4a1df5deabb1dceea

File: operators/src/pro/machine_learning/metadata_from_file.rs (hunk 2 above)
Comment (michaelmattig) [reply operators/src/pro/machine_learning/metadata_from_file.rs]: I removed the comment, as it originally targeted loading the model data in the operator, which now does use the database. 07374b850dd9a0d5638233a4a1df5deabb1dceea

We could still think about adding an API handler that autogenerates the MlMetadata, like we do for the OgrDataset. Currently, the user has to specify this manually when adding the model. But right now I think it only saves very little and adds another potential source of failure, if we use it by default.

File: services/src/machine_learning/name.rs (hunk 3 above)
Comment (michaelmattig) [reply services/src/machine_learning/name.rs]: I removed it 07374b850dd9a0d5638233a4a1df5deabb1dceea

File: operators/src/pro/machine_learning/metadata_from_file.rs (hunk 2 above)
Comment (jdroenner) [reply operators/src/pro/machine_learning/metadata_from_file.rs]: yes i guess it is file like this

**Review by jdroenner — APPROVED** [review]

**Pull request merged**